### PR TITLE
Add recoil rate scaling

### PR DESCRIPTION
**Pull request by snschune**
Unscaled recoil rates can lead to very large number of recoils that take long to assemble and may overrun available memory. This adds a common scaling factor for all `PKAGenerator` in the `Rasterizer` that is automatically considered when results are scaled.

### CODE REVIEW

File: src/userobjects/MyTRIMRasterizer.C [hunk 1]
```diff
@@ -356,7 +358,7 @@ MyTRIMRasterizer::finalize()
   if (_trim_parameters.desired_npka == 0 || _trim_parameters.desired_npka > _trim_parameters.original_npka)
   {
     _trim_parameters.scaled_npka = _trim_parameters.original_npka;
-    _trim_parameters.result_scaling_factor = 1;
+    _trim_parameters.result_scaling_factor = 1.0 / _trim_parameters.recoil_rate_scaling;
 
     if (_trim_parameters.desired_npka > _trim_parameters.original_npka)
       mooseDoOnce(mooseWarning("number_pka is larger than number of sampled PKAs and will be ignored!"));
```
Comment (dschwen) [inline src/userobjects/MyTRIMRasterizer.C]: Not quite part of this PR, but I don;t like this warning. We should name the parameter `max_pka_count` and drop the warning.

File: src/userobjects/MyTRIMRasterizer.C [hunk 2]
```diff
@@ -96,6 +96,7 @@ InputParameters validParams<MyTRIMRasterizer>()
   // Advanced options
   params.addParam<unsigned int>("interval", 1, "The time step interval at which TRIM BCMC is run");
   params.addParam<Real>("analytical_energy_cutoff", 0.0, "Energy cutoff in eV below which recoils are not followed explicitly but effects are calculated analytically.");
+  params.addParam<Real>("recoil_rate_scaling", 1.0, "A factor to scale computed reaction rates in the the PKAGenerator objects. This is useful to avoid extremely large PKA lists.");
```
Comment (dschwen) [inline src/userobjects/MyTRIMRasterizer.C]: I don't quite understand what the difference between the scaling factor and a cap on total PKA number is. Wouldn't the latter automatically trigger result scaling?

Comment (snschune) [reply src/userobjects/MyTRIMRasterizer.C]: the number of pka you want to run requires assembling the pka list. The pka list can be really long (`1e12` and up) which takes forever to assemble and may overrun memory. Even if you discard most of them, there is still a bottleneck there.

Comment (dschwen) [reply src/userobjects/MyTRIMRasterizer.C]: Ok, that makes sense. I still suggest we rename that other parameter.

**Comment by dschwen** [comment]
Can you rebase this? I factored out some PKAConstant stuff into a base class.

**Pull request merged**